### PR TITLE
Breaking backend optimizer addition

### PR DESCRIPTION
**Pull request by joonatank**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Update to backend https://github.com/City-of-Helsinki/tilavarauspalvelu-core/pull/1131
- Refactor (admin): remove reducers from Spaces and Resources.
- Remove (admin): custom tables from Spaces and Resources.
- Remove (admin): frontend sorting of ReservationUnits.
- Refactor (ui): move ui edit / cancel reservation queries to SSR.
- Refactor (admin): remove Joi (replaced with zod everywhere).
- Refactor (admin): remove auto fetching from recurring reservations (new endpoint gives the whole list).
- Refactor (admin): remove cache updates from reservation lists (refetch instead).

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of what's the fastest way to test your changes."

- Changes touching reservation units (so everything).
- Spaces and Resources editing refactored.
- Moving or creating new reservation. Collisions from space hierarchy have changed completely.

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)"

- [TILA-3238](https://helsinkisolutionoffice.atlassian.net/browse/TILA-3238)
- [TILA-2800](https://helsinkisolutionoffice.atlassian.net/browse/TILA-2800)

[TILA-2800]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-2800?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[TILA-3238]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3238?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: apps/ui/pages/reservations/[id]/cancel.tsx [hunk 1]
```diff
@@ -0,0 +1,99 @@
...
+    props: {
+      notFound: true,
+      ...commonProps,
+      ...(await serverSideTranslations(locale ?? "fi")),
+      key: `${pk}-cancel-${locale}`,
+    },
+  };
+}
+
+function ReservationParams(props: PropsNarrowed): JSX.Element {
```
Comment (joonatank) [inline apps/ui/pages/reservations/[id]/cancel.tsx:81]: rename page

File: apps/ui/pages/search/single.tsx [hunk 2]
```diff
@@ -176,42 +197,37 @@ const SearchSingle = (): JSX.Element => {
...
-    "reservationUnit-search",
-    {}
-  )[1];
 
-  const { data, fetchMore, loading, error, networkStatus } = useQuery<
+  const { data, fetchMore, error, networkStatus } = useQuery<
     Query,
     QueryReservationUnitsArgs
   >(RESERVATION_UNITS, {
+    // FIXME orderBy change
```
Comment (joonatank) [inline apps/ui/pages/search/single.tsx:205]: remove

File: packages/common/src/reservation-form/util.ts [hunk 3]
```diff
@@ -1,23 +1,38 @@
...
+} from "../../types/gql-types";
 import { reservationApplicationFields } from "./types";
+import { containsField } from "../metaFieldsHelpers";
 
-export const getReservationApplicationFields = ({
+// TODO this is bad
+// and there seems to be other functions that have same or similar names
+// what is the purpose and use cases for this one?
+// also it uses types that are exclusive to it
+// WHAT? why doesn't tsc catch if this isn't exported
```
Comment (joonatank) [inline packages/common/src/reservation-form/util.ts:13]: REMOVE

**Review by vincit-matu — APPROVED** [review]
...phew! 👍 Good stuff.

**Pull request merged**